### PR TITLE
refactor(robot-server): Robot server port rpc fast api

### LINKED ISSUE #5162: api: port rpc websocket support from api to robot-server
- [ ] Port business logic to use fastapi endpoints in robot-server
- [ ] Port api tests to robot-server using starlette's testing client

### PR DESCRIPTION
**Pull request by amitlissack**
## overview

This PR brings the RPC endpoint and websocket support to robot-server

## changelog

- brought rpc and serialize modules from api to robot-server along with tests.
- use starlette's websocket interface instead of aiohttp
- use starlette's testclient for unit tests.

## review requests

This is mostly a copy of code from api to robot-server with some changes to accomodate how starlette does websockets vs aiohttp.

## risk assessment

Very low / None. All code is behind a feature flag that is not active.

closes #5162

### CODE REVIEW

File: robot-server/robot_server/service/routers/rpc.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+from robot_server.service.dependencies import get_rpc_server
+from robot_server.service.rpc.rpc import RPCServer
+
+
+router = APIRouter()
+
+
+@router.websocket("/")
+async def websocket_endpoint(websocket: WebSocket,
```
Comment (sfoster1) [inline robot-server/robot_server/service/routers/rpc.py]: are we sure this is the right way to do the param here? `Depends()` will be called only once, at the time the interpreter parses the module. I'm not familiar with the way it works - is that ok?

Comment (amitlissack) [reply robot-server/robot_server/service/routers/rpc.py]: `Depends` does not call `get_rpc_server` on construction. It calls it when a request arrives. The pattern is correct.

**Comment by sfoster1** [comment]
> Regarding testing, i can do it on a raspberry pi but I don't have a robot.
> As I see it, these porting tickets are really about porting the business logic and tests. I see the rigid testing happening when we decide it's to switch over to fastapi from aiohttp. Because that rigid testing will happen then regardless of what testing we do now.

Maybe I'm overstating the rigidity of the testing. I'd still really like to run it on a robot and start, pause, and cancel a protocol. That should be pretty easy (I can do it at home).

>     * A wrapper would be hard. The rpc module is fully enmeshed with aiohttp. It would require extreme hackery.

Fair enough, especially given
>     * The git history can be found using `gl --follow robot_server/service/rpc/rpc.py`

`gl` here is an alias for `git log` right? Didn't know about that, thanks for letting me know.

>     * I get that's fragile and important, but it's not THAT complicated and precious. It's just code.

Haha fair enough, I've just had bad experiences with it before. I'll give it a test this morning.

**Comment by amitlissack** [comment]
> > Regarding testing, i can do it on a raspberry pi but I don't have a robot.
> > As I see it, these porting tickets are really about porting the business logic and tests. I see the rigid testing happening when we decide it's to switch over to fastapi from aiohttp. Because that rigid testing will happen then regardless of what testing we do now.
> 
> Maybe I'm overstating the rigidity of the testing. I'd still really like to run it on a robot and start, pause, and cancel a protocol. That should be pretty easy (I can do it at home).
> 
> > ```
> > * A wrapper would be hard. The rpc module is fully enmeshed with aiohttp. It would require extreme hackery.
> > ```
> 
> Fair enough, especially given
> 
> > ```
> > * The git history can be found using `gl --follow robot_server/service/rpc/rpc.py`
> > ```
> 
> `gl` here is an alias for `git log` right? Didn't know about that, thanks for letting me know.

Oops. Sorry. `gl` is my local alias for `git log` indeed. 
 
> > ```
> > * I get that's fragile and important, but it's not THAT complicated and precious. It's just code.
> > ```
> 
> Haha fair enough, I've just had bad experiences with it before. I'll give it a test this morning.

I appreciate that. We're all smart folks. We should be afraid of no code!

**Pull request merged**